app/gui/src/project-view/providers/globalEventRegistry.ts

```diff
@@ -0,0 +1,63 @@
+import type { WindowEventTarget } from '@/composables/events'
+import { createContextStore } from '@/providers'
+import { identity } from '@vueuse/core'
+
+interface GlobalEventRegistry {
+  globalEventRegistry: WindowEventTarget
+  globalEventRegistryPre: WindowEventTarget
+}
+
+const [provideRegistry, useGlobalEventRegistry] = createContextStore(
+  'GlobalEvent',
+  identity<GlobalEventRegistry>,
+)
+
+export { useGlobalEventRegistry }
+
+/**
+ * Create a {@link GlobalEventRegistry} and make is available to the component's children.
+ * @returns the created registry
+ */
+export function provideGlobalEventRegistry(): GlobalEventRegistry {
+  const globalEventRegistryPre = eventRegistry()
+  const globalEventRegistry = eventRegistry(window, globalEventRegistryPre)
+  const registry = { globalEventRegistry, globalEventRegistryPre }
+  provideRegistry(registry)
+  return registry
+}
```

Comment:
We could make use of createContextStore "factory" parameter:

```suggestion
export const [provideGlobalEventRegistry, useGlobalEventRegistry] = createContextStore(
  'GlobalEvent',
  () => {
    const globalEventRegistryPre = eventRegistry()
    const globalEventRegistry = eventRegistry(window, globalEventRegistryPre)
    return { globalEventRegistry, globalEventRegistryPre }
  },
)
```

Also, here should be a documentation of GlobalEventRegistry